resources/lambda-layers.go

```diff
@@ -0,0 +1,75 @@
+package resources
+
+import (
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/rebuy-de/aws-nuke/pkg/types"
+)
+
+type lambdaLayer struct {
+	svc       *lambda.Lambda
+	layerName *string
+	version   int64
+}
+
+func init() {
+	register("LambdaLayer", ListLambdaLayers)
+}
+
+func ListLambdaLayers(sess *session.Session) ([]Resource, error) {
+	svc := lambda.New(sess)
+
+	layers := make([]*lambda.LayersListItem, 0)
+
+	params := &lambda.ListLayersInput{}
+
+	err := svc.ListLayersPages(params, func(page *lambda.ListLayersOutput, lastPage bool) bool {
+		for _, out := range page.Layers {
+			layers = append(layers, out)
+		}
+		return true
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	resources := make([]Resource, 0)
+
+	for _, layer := range layers {
+
+		var i int64 = int64(1) // is there a better way to do this
+		for i <= *layer.LatestMatchingVersion.Version {
+			resources = append(resources, &lambdaLayer{
```

Comment:
I don't understand this. Does lambda always have *all* versions from 1 to `LatestMatchingVersion`?

Reply:
It did not throw an error when it tried to delete a version that did not exist (I tested that), but on reflection the right way is to use the list provided by `listLayerVersions`.  I'll update and resubmit.

Reply:
Thanks. It is not really about errors, but about performance for cases where there is a high version number.